src/org/opensolaris/opengrok/configuration/RuntimeEnvironment.java

```diff
@@ -1970,7 +2028,7 @@ public SuperIndexSearcher getIndexSearcher(String proj) throws IOException {
      * change so we go through the SearcherManager objects and close those where
      * the corresponding project is no longer present.
      */
-    private void refreshSearcherManagerMap() {
+    synchronized public void refreshSearcherManagerMap() {
```

Comment:
should not be necessary (in the worst case the iterator will not reflect current state), removed.